test/e2e/test-plugins/plugins-api/sidebar.js

```diff
@@ -48,6 +50,24 @@
 					},
 					__( 'Reset' )
 				)
+			),
+			el(
+				Button,
+				{
+					isPrimary: true,
+					onClick: () => {
+						dispatch( 'core/editor' ).addAnnotation( {
+							source: 'e2e-tests',
+							blockClientId: select( 'core/editor' ).getBlockOrder()[ 0 ],
+							richTextIdentifier: 'content',
+							range: {
+								start: 9,
```

Comment:
How does a plugin annotate the current selection in a rich text field?

Reply:
I do it like this in the annotations-tester plugin:

```js
const selection = window.getSelection();
const range = selection.rangeCount ? selection.getRangeAt( 0 ) : null;
const editable = range ? getClosestEditable( range.startContainer ) : null;

const record = wp.richText.create( { element: editable, range } );
const { start, end } = record;
```

We could offer an action to annotate the current selection.

Reply:
This feels like something that we should provide an API for. Something in the editor module. `wp.select( 'core/editor' ).getCurrentRichTextSelection()`

and it returns something like:

```js
{
   richTextIdentifier, blockClientUid, range
}
```

Reply:
The way I see to make `richTextIdentifier` available is if we keep track of the selected `RichText` in the state. Do you think we should do that here? Or do you see a different approach to solve the same problem? The conceptual issue I run into is that the selector can't know what the `richTextIdentifier` is.

Reply:
yes, that's implicitly what I'm proposing here. Keep the selection (richTextIdentifier and range) in the `core/editor` store.

Reply:
Agreed that there should be an API to apply a format to the current range in the current selected RichText instance. It's much more reliable than using the browser API.

Reply:
@youknowriad Let me know if this is what you had in mind: https://github.com/WordPress/gutenberg/pull/7718/commits/1eb6433e08e4a03f9fa2c18592789cde1160de0d.

Reply:
yes something like that. I think the `blockClientId` is not necessary in the action because it's already tracked in the selection reducer. Also, I think we should handle `onBlur` to reset the selection to `null` and in the docs we may want to check if there's a selection before calling `addAnnotation`

Reply:
`onBlur` might get tricky as soon as you start using toolbar controls. I'm also not quite sure if it wouldn't fail the e2e test added which exposes button. It might reset selection when you click the button :)

Reply:
Yes, I had purposefully not implemented `onBlur`, because of those problems. I was also thinking that technically the last `RichText` selection is in the state this way, which shouldn't be a problem and it would allow keyboard users to annotate the 'current' selection.

Reply:
> I think the blockClientId is not necessary in the action because it's already tracked in the selection reducer.

I don't know how this would work (is it always the `start` block from the blockSelection?), but I can do this investigation. It doesn't block this PR though, because of the nice architecture such that the internal state isn't exposed we can refactor this later. I've created an issue to track this: https://github.com/WordPress/gutenberg/issues/11615

> in the docs we may want to check if there's a selection before calling addAnnotation

This is not necessary, because the annotations reducer rejects all annotations with a `range` scope that don't have a valid `start` and `end`. So for the simplicity of the docs we can leave this out.